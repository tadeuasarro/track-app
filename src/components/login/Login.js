import Navbar from '../navbar/Navbar';

const Login = () => (
  <div>
    <Navbar />
    Some login form
  </div>
);

export default Login;
