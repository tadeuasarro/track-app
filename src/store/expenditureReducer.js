import { SET_EXPENDITURES } from '../actions/expenditure';
import filterExpenditures from '../helpers/filterExpenditures';

const initialState = {
  expenditures: [],
  summary: false,
};

const expenditureReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_EXPENDITURES:
      return {
        expenditures: action.payload,
        summary: filterExpenditures(action.payload),
      };
    default:
      return state;
  }
};

export default expenditureReducer;
