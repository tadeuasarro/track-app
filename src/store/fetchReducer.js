import { SET_LOADING, SET_ERRORS } from '../actions/fetch';

const initialState = {
  loading: false,
  errors: false,
};

const fetchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_LOADING:
      return {
        loading: true,
        errors: false,
      };
    case SET_ERRORS:
      return {
        loading: false,
        errors: action.payload,
      };
    default:
      return state;
  }
};

export default fetchReducer;
