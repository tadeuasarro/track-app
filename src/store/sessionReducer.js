import { SET_CURRENT_USER, UPDATE_USER_TARGET } from '../actions/session';

const initialState = {
  username: false,
  id: false,
  target: false,
  token: false,
};

const sessionReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        username: action.payload.user.username,
        id: action.payload.user.id,
        target: action.payload.user.target,
        token: action.payload.token,
      };
    case UPDATE_USER_TARGET:
      return {
        ...state,
        target: action.payload,
      };
    default:
      return state;
  }
};

export default sessionReducer;
