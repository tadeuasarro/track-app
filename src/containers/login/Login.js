import { useDispatch } from 'react-redux';
import { useState } from 'react';
import { setCurrentUser } from '../../actions/session';
import createUser from '../../api/createUser';
import createSession from '../../api/createSession';
import Error from '../../components/error/Error';
import './login.css';

const Login = () => {
  const dispatch = useDispatch();

  const [state, setState] = useState({
    pending: false,
    error: false,
  });

  const handleClick = async key => {
    const username = document.getElementById('login-input').value;

    setState({
      ...state,
      pending: true,
    });

    let res = {};

    if (key) {
      const target = 0;
      res = await createUser({ username, target });
    } else {
      res = await createSession(username);
    }

    setState(res.state);

    if (!res.state.error) {
      dispatch(setCurrentUser(res.payload));
    }

  };

  const errorObj = (!state.error ? {} : state.error);

  return (
    <div className="login-container">
      <div className="instructions">
        <p>Welcome to Personal Expenses, your personal expenses managing application.</p>
        <p>Login with your account, or fill the form and click on register!</p>
      </div>
      <form className="login-form">
        <input id="login-input" placeholder="Username" className="login-input" type="text" />
        <button onClick={() => handleClick(false)} className="login-button" type="button">Login</button>
        <Error error={errorObj.login} />
        <button onClick={() => handleClick(true)} className="register-button" type="button">Register</button>
        <Error error={errorObj.username} />
      </form>
    </div>
  );
};

export default Login;
