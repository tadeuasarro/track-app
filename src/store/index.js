import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import sessionReducer from './sessionReducer';
import expenditureReducer from './expenditureReducer';
import fetchReducer from './fetchReducer';

const store = createStore(
  combineReducers({
    session: sessionReducer,
    expenditure: expenditureReducer,
    fetch: fetchReducer,
  }),
  applyMiddleware(thunk),
);

export default store;
