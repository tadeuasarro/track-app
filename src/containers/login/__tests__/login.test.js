import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/index';
import Login from '../Login';

describe('Rendering component', () => {
  it('creates an App component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Login />
          </Provider>
        </BrowserRouter>,
      );
    });
    // eslint-disable-next-line
    const element = screen.getByText('Login');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
