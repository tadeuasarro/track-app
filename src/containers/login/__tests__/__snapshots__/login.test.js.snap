// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="login-container"
>
  <div
    className="instructions"
  >
    <p
      className="welcome-text"
    >
      Welcome to Personal Expenses, your personal expenses managing application.
    </p>
    <p
      className="welcome-text"
    >
      Login with your account, or fill the form and click on register!
    </p>
  </div>
  <form
    className="login-form"
  >
    <input
      className="login-input"
      id="login-input"
      onChange={[Function]}
      placeholder="Username"
      type="text"
    />
    <button
      className="login-button"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    
    <button
      className="register-button"
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
    
  </form>
</div>
`;
